stubs/synapse/synapse_rust/push.pyi

```diff
@@ -55,6 +56,8 @@ class PushRuleEvaluator:
     def __init__(
         self,
         flattened_keys: Mapping[str, str],
+        user_mentions: Set[str],
```

Comment:
Would `AbstractSet` make sense here, to permit passing both frozensets and sets to this function?

Reply:
It doesn't work properly with pyo3 when I tested.

Reply:
For more info, if I default to using a `frozenset()` if e.g. the `m.mentions` field doesn't exist, then we throw this error when attempting to pass that to Rust:

```
	Traceback (most recent call last):
	  File "/Users/clokep/matrix/synapse/synapse/http/server.py", line 307, in _async_render_wrapper
	    callback_return = await self._async_render(request)
	  File "/Users/clokep/matrix/synapse/synapse/http/server.py", line 513, in _async_render
	    callback_return = await raw_callback_return
	  File "/Users/clokep/matrix/synapse/synapse/rest/client/room.py", line 162, in on_POST
	    info, _ = await self._room_creation_handler.create_room(
	  File "/Users/clokep/matrix/synapse/synapse/handlers/room.py", line 926, in create_room
	    ) = await self._send_events_for_new_room(
	  File "/Users/clokep/matrix/synapse/synapse/handlers/room.py", line 1148, in _send_events_for_new_room
	    ev = await self.event_creation_handler.handle_new_client_event(
	  File "/Users/clokep/matrix/synapse/synapse/util/metrics.py", line 113, in measured_func
	    r = await func(self, *args, **kwargs)
	  File "/Users/clokep/matrix/synapse/synapse/handlers/message.py", line 1433, in handle_new_client_event
	    result, _ = await make_deferred_yieldable(
	  File "/Users/clokep/.pyenv/versions/synapse/lib/python3.9/site-packages/twisted/internet/defer.py", line 1697, in _inlineCallbacks
	    result = context.run(gen.send, result)
	  File "/Users/clokep/matrix/synapse/synapse/handlers/message.py", line 1469, in _persist_events
	    await self._bulk_push_rule_evaluator.action_for_events_by_user(
	  File "/Users/clokep/matrix/synapse/synapse/push/bulk_push_rule_evaluator.py", line 287, in action_for_events_by_user
	    await self._action_for_event_by_user(event, context, event_id_to_event)
	  File "/Users/clokep/matrix/synapse/synapse/util/metrics.py", line 113, in measured_func
	    r = await func(self, *args, **kwargs)
	  File "/Users/clokep/matrix/synapse/synapse/push/bulk_push_rule_evaluator.py", line 366, in _action_for_event_by_user
	    evaluator = PushRuleEvaluator(
	TypeError: argument 'user_mentions': 'frozenset' object cannot be converted to 'PySet'
```

Note that we could *always* use `frozenset` if we wanted, but I also wasn't really able to get that to work properly with pyo3.